File: 240311/최고의 33위치/best-place-of-33.cpp
#include <bits/stdc++.h>
using namespace std;

const int dy[9] = {0, 0, 0, 1, 1, 1, 2, 2, 2};
const int dx[9] = {0, 1, 2, 0, 1, 2, 0, 1, 2};

int n, arr[21][21], ans; 

void input() {
    cin >> n;

    for (int i=1; i<=n; i++) {
        for (int j=1; j<=n; j++) {
            cin >> arr[i][j];
        }
    }
}

void solve() {
    for (int i=1; i<=n-2; i++) {
        for (int j=1; j<=n-2; j++) {
            int cnt = 0;
            for (int k=0; k<9; k++) {
                int y = i + dy[k];
                int x = j + dx[k];
                cnt += arr[y][x];
            }
            ans = max(ans, cnt);
        }
    }
    cout << ans;
}

int main() {
    input();
    solve();
    return 0;
}

File: 240408/최고의 33위치/best-place-of-33.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");

const n = Number(input[0]);
const arr = input.slice(1, 1 + n).map(row => row.split(" ").map(Number));

const dy = [0, 0, 0, 1, 1, 1, 2, 2, 2];
const dx = [0, 1, 2, 0, 1, 2, 0, 1, 2];

let ans = 0;
for (let i = 0; i < n - 2; i++) {
    for (let j = 0; j < n - 2; j++) {
        let cnt = 0;
        for (let k = 0; k < 9; k++) {
            let ny = j + dy[k];
            let nx = i + dx[k];
            if (arr[ny][nx] === 1) {
                cnt++;
            }
        }
        ans = Math.max(ans, cnt);
    }
}

console.log(ans);